src/formatDate.js

```diff
@@ -7,8 +7,57 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const dateParts = date.split(fromSeparator);
+
+  const indexes = {
+    YYYY: fromFormat.indexOf('YYYY'),
+    YY: fromFormat.indexOf('YY'),
+    MM: fromFormat.indexOf('MM'),
+    DD: fromFormat.indexOf('DD'),
+  };
+
+  const newDate = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const part = toFormat[i];
+
+    if (part === 'YYYY') {
+      if (dateParts[indexes['YYYY']] !== undefined) {
+        newDate.push(dateParts[indexes['YYYY']]);
+      } else {
+        newDate.push(convertYear(dateParts[indexes['YY']], 'YY'));
+      }
+    } else if (part === 'YY') {
+      if (dateParts[indexes['YY']] !== undefined) {
+        newDate.push(dateParts[indexes['YY']]);
+      } else {
+        newDate.push(convertYear(dateParts[indexes['YYYY']], 'YYYY'));
+      }
+    } else if (indexes[part] !== -1) {
+      newDate.push(dateParts[indexes[part]]);
+    }
+  }
+
+  const toSeparator = toFormat[toFormat.length - 1];
+
+  return newDate.join(toSeparator);
+}
+
+function convertYear(year, format) {
+  if (format === 'YY') {
+    const yy = Number(year);
+
+    if (yy < 30) {
+      return '20' + year;
+    } else {
+      return '19' + year;
+    }
+  } else if (format === 'YYYY') {
+    return year.slice(-2);
+  }
 }
```

Comment:
The `convertYear` function does not correctly handle the conversion from 'YY' to 'YYYY'. It should use the last two digits of the given year when converting from 'YYYY' to 'YY', and it should prepend '20' or '19' based on the value of the last two digits when converting from 'YY' to 'YYYY'. The current implementation is incorrect because it uses the entire year string when converting from 'YY' to 'YYYY'.